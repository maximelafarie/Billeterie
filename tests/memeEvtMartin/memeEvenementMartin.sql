-- ============================================================================
-- Fichier          : memeEvenementMartin.sql
-- Auteur           : Maxime LAFARIE
-- Date             : Janvier 2014
-- Base				: Billeterie
-- Projet           : Billeterie
-- Role             : Clients ayant acheté un ou des billets pour le même 
--					  évènement que Jean MARTIN le 15/12/2013.
-- ============================================================================

SELECT C2.idcli, C2.nomcli, C2.prenomcli
FROM CLIENT C1, CLIENT C2, EVENEMENT E1, EVENEMENT E2, BILLET B1, BILLET B2
WHERE C1.nomcli = 'MARTIN'
AND C1.idcli = B1.idcli
AND E1.dateevt = '15/12/2013'
AND B1.idevt = E1.idevt
AND E1.idevt = E2.idevt
AND E1.dateevt = E2.dateevt
AND B2.idevt = E2.idevt
AND C2.idcli = B2.idcli
AND C2.nomcli != 'MARTIN'
ORDER BY C2.idcli ASC;