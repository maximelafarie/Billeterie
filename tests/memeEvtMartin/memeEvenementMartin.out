fichier resultat du test : memeEvenementMartin.out
Contenu de la base :
SQL> @ utilitaires/afficherContenuBaseBilleterie.sql
SQL> -- ==========================
SQL> -- fichier : afficherContenuBaseBilleterie.sql
SQL> -- base : BILLETERIE
SQL> -- auteur(s) : Maxime LAFARIE
SQL> -- auteur(s) : Nicolas DOUSSON
SQL> -- date : 07/01/14
SQL> -- role : Afficher le contenu des differentes tables de la base
SQL> -- projet : Billeterie
SQL> -- ==========================
SQL> 
SQL> prompt Contenu de la table client :
Contenu de la table client :
SQL> 
SQL> SELECT * FROM CLIENT;

     IDCLI NOMCLI          PRENOMCLI       TELCLI     MAILCLI                   
---------- --------------- --------------- ---------- --------------------      
ADRCLI                                                                          
---------------                                                                 
         1 MARTIN          Jean            0505050505 martin@gmail.com          
La Rochelle                                                                     
                                                                                
         2 DUPONT          Michel          0605050505 michel@gmail.com          
Niort                                                                           
                                                                                

SQL> 
SQL> prompt Contenu de la table billet :
Contenu de la table billet :
SQL> 
SQL> SELECT * FROM BILLET;

  IDBILLET CODEBILLET           IDCLI    IDTARIF IDSITERESA      IDEVT          
---------- --------------- ---------- ---------- ---------- ----------          
         1 15ED68EP10EPD            1          1          1          1          
         2 85PO36DM19CBD            1          3                     2          
         3 89X736DM55CZM            2          1                     1          

SQL> 
SQL> prompt Contenu de la table evenement :
Contenu de la table evenement :
SQL> 
SQL> SELECT * FROM EVENEMENT;

     IDEVT TITREEVT        DATEEVT    TYPEEVT    IDSALLE                        
---------- --------------- ---------- ------- ----------                        
         1 Les Joyaux      01-02-2014 sport            1                        
         2 Clipto          10-04-2014 concert          2                        

SQL> 
SQL> prompt Contenu de la table salle :
Contenu de la table salle :
SQL> 
SQL> SELECT * FROM SALLE;

   IDSALLE NOMSALLE        ADRSALLE                                             
---------- --------------- --------------------                                 
         1 CARLINE         La Rochelle                                          
         2 Melrane         Bordeaux                                             

SQL> 
SQL> prompt Contenu de la table tarif :
Contenu de la table tarif :
SQL> 
SQL> SELECT * FROM TARIF;

   IDTARIF       PRIX TYPETARI                                                  
---------- ---------- --------                                                  
         1        100 plein                                                     
         3         85 plein                                                     

SQL> 
SQL> prompt Contenu de la table siteresa :
Contenu de la table siteresa :
SQL> 
SQL> SELECT * FROM SITERESA;

IDSITERESA NOMSITERESA          URLSITERESA                                     
---------- -------------------- --------------------                            
         1 LeMoinsCher          www.lmc.com                                     

SQL> 
SQL> -- ============================================================================
SQL> -- Requete a tester
SQL> -- ============================================================================
SQL> 
SQL> -- Debut de l'affichage du resultat de la requete sur le client SqlPlus
SQL> prompt 'Resultat de la requete'
'Resultat de la requete'
SQL> @ tests/memeEvenementMartin/memeEvenementMartin.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : memeEvenementMartin.sql
SQL> -- Auteur		 : Maxime LAFARIE
SQL> -- Date		 : Janvier 2014
SQL> -- Base			     : Billeterie
SQL> -- Projet		 : Billeterie
SQL> -- Role		 : Clients ayant acheté un ou des billets pour le même
SQL> -- 				       évènement que Jean MARTIN le 15/12/2013.
SQL> -- ============================================================================
SQL> 
SQL> SELECT C2.idcli, C2.nomcli, C2.prenomcli
  2  FROM CLIENT C1, CLIENT C2, EVENEMENT E1, EVENEMENT E2, BILLET B1, BILLET B2
  3  WHERE C1.nomcli = 'MARTIN'
  4  AND C1.idcli = B1.idcli
  5  AND E1.dateevt = '15/12/2013'
  6  AND B1.idevt = E1.idevt
  7  AND E1.idevt = E2.idevt
  8  AND E1.dateevt = E2.dateevt
  9  AND B2.idevt = E2.idevt
 10  AND C2.idcli = B2.idcli
 11  AND C2.nomcli != 'MARTIN'
 12  ORDER BY C2.idcli ASC;

no rows selected

SQL> 
SQL> -- Fin de l'affichage sur le client SqlPlus
SQL> SET echo OFF
